examples/blockdiagperm.cc

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+
+#include "tc/aten/aten_compiler.h"
+#include "tc/autotuner/genetic_autotuner_aten.h"
+#include "tc/core/cuda/cuda.h"
+#include "tc/core/cuda/cuda_tc_executor.h"
+#include "tc/core/flags.h"
+#include "tc/core/mapping_options.h"
+
+DEFINE_string(tuner_proto, "", "Filename to load and store proto cache ");
```

Comment:
this appears in every example, maybe you want an actual tuner flag instead?

Reply:
This was removed in a PR by @ttheodor because it was dead code in the mapper and gave a false impression that it was doing something in the mapper. In practice each binary may or may not want to save. For the simple examples we build small reinforcement loops by using the proto and running multiple times. One could argue that that this belongs in a harness, sure but then other things too and we end up in a position where people complain about complexity (e.g. benchmarks).  So this is our "examples" tradeoff, we may want to avoid the reinforcement loop in the future if we are picky about the flag.